packages/wrangler/src/inspect.ts

```diff
@@ -605,20 +628,28 @@ export default function useInspector(props: InspectorProps) {
 			);
 			try {
 				// Intercept Debugger.scriptParsed responses to inject URL schemes
-				const message = JSON.parse(event.data as string);
-				if (
-					message.method === "Debugger.scriptParsed" &&
-					// Breakpoint debugging doesn't work (breakpoints can be set, but not hit) with the worker:// scheme, so
-					// disable in local mode.
-					!localMode
-				) {
-					// Add the worker:// scheme conditionally, since some module types already have schemes (e.g. wasm)
-					message.params.url = new URL(
-						message.params.url,
-						`worker://${props.name}`
-					).href;
-					localWebSocket.send(JSON.stringify(message));
-					return;
+				if (localWebSocket.isDevTools) {
+					const message = JSON.parse(event.data as string);
+					if (message.method === "Debugger.scriptParsed") {
+						if (message.params.sourceMapURL) {
+							const url = new URL(
+								message.params.sourceMapURL,
+								message.params.url
+							);
+							if (url.protocol === "file:") {
+								// `sourceMaps.current` is always defined after `useRef()`
+								assert(sourceMaps.current !== undefined);
+								const name = props.name ?? "worker";
+								const id = crypto.randomUUID();
+								sourceMaps.current.set(id, fileURLToPath(url));
+								// The hostname of this URL will show up next to the cloud icon
+								// under authored sources, so use the worker name.
+								message.params.sourceMapURL = `worker://${name}/${id}`;
```

Comment:
Does this not show the file name as `id` in the deployed sources section? How does this work for scripts without a sourcemap?